### DISCUSSION TITLE
Unable to connect to netsuite via odbc or jdbc

### DISCUSSION THREAD

**Discussion by D3SL** [dbeaver/dbeaver#11999] [Q&A]
#### System information: 
- Win10 Pro 18363.900
- DBeaver 7.1.2

#### Connection specification:
- Netsuite's ODBC 8.10.92.0 and JDBC 8.10.85.0

#### Describe your question:
I'm attempting to connect to netsuite's internal database (essentially oracle SQL) via their ODBC or JDBC drivers. I've got the ODBC DSN set up in windows and can connect fine via R and Access, but every attempt to create a DB connection in DBeaver via ODBC or creating a custom driver using netsuite's JDBC driver produces a "no suitable driver found" error.

Opening this as a question rather than bug because I'm not confident this isn't user error on my part.

**Comment by uslss** [comment]
Hi @D3SL , probably it's some driver config issues. Could you post a screenshot of your driver config?

Possibly, it'll be useful for you https://github.com/dbeaver/dbeaver/wiki/Database-drivers

**Comment by D3SL** [comment]
Sure thing @uslss , here's the current setup for the custom "netsuite" driver I created and the built in ODBC driver. I've connected successfully in other programs so I know the issue has to be somewhere inside DBeaver, either a genuine bug or something I'm setting up wrong.

![dbeaver netsuite and odbc drivers](https://user-images.githubusercontent.com/63225966/88115074-036e3380-cbbe-11ea-8308-8111333f94f9.png)

**Comment by D3SL** [comment]
@uslss I just saw this was changed to bug. If it's relevant I got this to half work _once_ on my very first attempt using DBeaver's ODBC driver. It said that it connected rather than giving me the error but then would simply hang and never allow me to browse the list of tables or data like it does with a normal working connection. After updating DBeaver I wound up getting the responses detailed above.

**Comment by Sizzlak** [comment]
If you untick "use legacy JDBC instantiation" it's not complaining about "no suitable driver found" but you'll get the error "TCP/IP error, connection reset by peer." 
I'm able to connect via other programs like TD SQLA or DBVis.

**Comment by uslss** [comment]
Could you check if the connection works without any changes in DBeaver default ODBC driver?
If it doesn't, please provide an error message.
Thanks in advance

**Comment by D3SL** [comment]
> Could you check if the connection works without any changes in DBeaver default ODBC driver?
> If it doesn't, please provide an error message.
> Thanks in advance

As stated in the initial post I tried that as well and received the error "no suitable driver found":

```
eclipse.buildId=unknown
java.version=11.0.5
java.vendor=AdoptOpenJDK
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Command-line arguments:  -os win32 -ws win32 -arch x86_64

org.jkiss.dbeaver.model
Error
Sun Sep 13 10:15:51 IDT 2020
No suitable driver found for jdbc:odbc:NetSuite
```

**Comment by uslss** [comment]
Seems that x32 ODBC DSN setup was used.
Dbeaver supports only x64 odbc connections

**Comment by Jerr31** [comment]
Is anybody able to make a connection via ODBC 64-bit to NetSuite?
